### PR TITLE
New music and loading animation

### PR DESCRIPTION
**Pull request by NathanHinthorne**
* Added new music to the music folder.
* Added a way to access and play the music in Util.
* Added load animation to be played while asset manager is in the middle of loading audio/images in its cache (not completed yet)

### CODE REVIEW

File: js/classes/asset_manager.js [hunk 1]
```diff
@@ -42,14 +42,19 @@ class AssetManager {
             const path = this.downloadQueue[i];
             console.log(path);
 
+            // make sure the path is a string
+            if (typeof path !== 'string') {
```
Comment (b1gd3vd0g) [inline js/classes/asset_manager.js:46]: Nice! More errors with descriptive strings makes debugging easier!

File: js/classes/entities/foreground/loading_animation.js
Comment (b1gd3vd0g) [inline js/classes/entities/foreground/loading_animation.js:1]: I like the effort put into this class. It does a lot of things that I wouldn't have thought of.

**Review by b1gd3vd0g — COMMENTED** [review]
In JavaScript, it is normal to have too many semicolons. They belong after function bodies and classes - your code doesn't seem to have this (which is fine, it works, just a bit off convention)

File: js/scripts/main.js
Comment (b1gd3vd0g) [inline js/scripts/main.js:1]: Don't do this! Never commit your main file with the default zone switched - this will force everybody to your zone!

**Review by b1gd3vd0g — APPROVED** [review]

File: js/classes/entities/foreground/loading_animation.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+class LoadingAnimation {
+    constructor() {
+        const center = new Vector(CANVAS.width / 2, CANVAS.height / 2);
+        
+        this.circles = [
+            { x: center.x - 50, y: center.y, dy: 20},
```
Comment (calebkrauter) [inline js/classes/entities/foreground/loading_animation.js:6]: Can you explain how this works?

File: js/classes/entities/foreground/loading_animation.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        // setInterval() is the reason GameEngine does not control the update-render loop.
+
+        // setInterval() repeatedly calls a function or executes a code snippet, 
+        // with a fixed time delay between each call.
+        // https://developer.mozilla.org/en-US/docs/Web/API/setInterval
+        this.intervalId = setInterval(() => this.update(), 100);
+    }
+
+    stop() {
+        if (this.intervalId) {
```
Comment (calebkrauter) [inline js/classes/entities/foreground/loading_animation.js:24]: What does this block of code mean?

Comment (calebkrauter) [reply js/classes/entities/foreground/loading_animation.js:24]: What is intervalId

File: js/classes/entities/foreground/loading_animation.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    update() {
+        // Clear the canvas
+        CTX.clearRect(0, 0, CANVAS.width, CANVAS.height);
+
+        // Update and draw each circle
+        for (let i = 0; i < this.circles.length; i++) {
+            let circle = this.circles[i];
+
+            // If this is the current circle, make it bounce and squish
+            if (i === this.currentCircle) {
```
Comment (calebkrauter) [inline js/classes/entities/foreground/loading_animation.js:39]: Very nice.

File: js/classes/entities/foreground/loading_animation.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                circle.y += circle.dy;
+                if (circle.y > LoadingAnimation.LOWER_BOUND || circle.y < LoadingAnimation.HIGHER_BOUND) {
+                    circle.dy *= -1;
+                }
+            }
+
+            // Draw the circle
```
Comment (calebkrauter) [inline js/classes/entities/foreground/loading_animation.js:46]: Nice work with the loading animation. Looks cool.

File: js/classes/entities/foreground/loading_animation.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            CTX.lineWidth = 3;
+            CTX.strokeStyle = 'black';
+            CTX.stroke();
+        }
+
+        // Move to the next circle
+        this.currentCircle = (this.currentCircle + 1) % this.circles.length;
+    }
+
+    static get LOWER_BOUND() {
```
Comment (calebkrauter) [inline js/classes/entities/foreground/loading_animation.js:62]: What are these bounds for? I'm confused on how the height plus or minus 50 constrains the circles to the center of the screen if that is what it's doing. Are these even being used?

File: js/classes/entities/midground/weapon/slingshot.js [hunk 7]
```diff
@@ -72,17 +74,23 @@ class Slingshot {
...
     }
 
     update() {
+        if (this.shootTimer > 0) {
+            this.shootTimer -= GAME.clockTick;
+        }
         if (GAME.user.aiming) {
             this.aim();
-        } else if (GAME.user.firing) {
+        } else if (GAME.user.firing && this.shootTimer <= 0) {
```
Comment (calebkrauter) [inline js/classes/entities/midground/weapon/slingshot.js]: The delay is great!

**Review by calebkrauter — APPROVED** [review]
Looks great!

File: js/scripts/main.js
Comment (NathanHinthorne) [reply js/scripts/main.js:1]: Got it. Thanks for catching that @B1GD3VD0G

File: js/classes/entities/foreground/loading_animation.js
Comment (NathanHinthorne) [reply js/classes/entities/foreground/loading_animation.js:1]: Thank you!

File: js/classes/entities/foreground/loading_animation.js (hunk 3 above)
Comment (NathanHinthorne) [reply js/classes/entities/foreground/loading_animation.js:24]: that was bad naming on my part. It's just a reference to the interval itself.

File: js/classes/entities/foreground/loading_animation.js (hunk 2 above)
Comment (NathanHinthorne) [reply js/classes/entities/foreground/loading_animation.js:6]: Each of the circles are updated one at a time, but for that they need these individual properties

File: js/classes/entities/foreground/loading_animation.js (hunk 6 above)
Comment (NathanHinthorne) [reply js/classes/entities/foreground/loading_animation.js:62]: Whoops, good catch. I meant to divide the height by 2 for each of these.

**Pull request merged**